src/classes/UTIL_OrgTelemetry_BATCH.cls

```diff
@@ -0,0 +1,82 @@
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2018
+* @group Utilities
+* @description A more controllable method to handle the asynchronous submission of Telemetry data to the LMO
+* without having to worry about governor limits when it comes to querying large datasets in LDV orgs. By iterating
+* through a custom List<> of Enum Types with a batch size of ONE, the execute() method can call a service method
+* to handle querying and submitting the specific telemetry data only for that Enum type. If for some reason that
+* one query were to fail, which is theoretically possible in a huge organization where there may be other jobs
+* executing or other factors at play, the rest of this job will continue. That ensures that our LMO collects as much
+* data as possible. In addition, it can help us identify which specific telemetry data processes are failing.
+*/
+public without sharing class UTIL_OrgTelemetry_BATCH implements Database.Batchable<UTIL_OrgTelemetry_SVC.TelemetryType> {
+
+    /**
+     * @description List of High Level Telemetry Data Types to be processed asynchronously by this batch job.
+     */
+    private final List<UTIL_OrgTelemetry_SVC.TelemetryType> telemetryTypes =
+        new List<UTIL_OrgTelemetry_SVC.TelemetryType>{
+            UTIL_OrgTelemetry_SVC.TelemetryType.NPSP_Settings,
+            UTIL_OrgTelemetry_SVC.TelemetryType.Org_Environment,
+            UTIL_OrgTelemetry_SVC.TelemetryType.Data_MaxNumRelatedOpps,
+            UTIL_OrgTelemetry_SVC.TelemetryType.Data_Count_of_RD_Opps_All,
+            UTIL_OrgTelemetry_SVC.TelemetryType.Data_Count_of_RD_Opps_OpenEnded,
+            UTIL_OrgTelemetry_SVC.TelemetryType.Data_CountOfErrorLog
+        };
+
+    /**
+     * @description Return the list of Telemetry Type Enum values to be processed by this batch job
+     * @param BC BatchableContext (not used)
+     * @return telemetryTypesList list from above
```

Comment:
The argument needs to be updated to `Iterable<UTIL_OrgTelemetry_SVC.TelemetryType>`